ClearKeep/Scenes/Home/HomeView.swift

```diff
@@ -9,96 +9,323 @@ import SwiftUI
+	}
+
+	@State private(set) var loadableUrl: Loadable<Bool> = .notRequested {
+		didSet {
+			switch loadableUrl {
+			case .loaded:
+				isLoading = false
+				customServer.customServerURL = serverURL
+				customServer.isSelectedCustomServer = true
+				navigateToLogin = true
+			case .failed(let error):
+				isLoading = false
+				self.error = HomeErrorView(error)
+				self.isShowError = true
+			case .isLoading:
+				isLoading = true
+			default: break
+			}
+		}
 	}
+	@StateObject var callViewModel: CallViewModel = CallViewModel()
 	
+	@State private(set) var servers: [ServerViewModel] = []
+	@State private(set) var searchKeyword: String = ""
+	@State private(set) var searchInputStyle: TextInputStyle = .default
+	@State private(set) var isShowMenu: Bool = false
+	@State private(set) var isAddNewServer: Bool = false
+	@State private(set) var groups: [GroupViewModel] = []
+	@State private(set) var peers: [GroupViewModel] = []
+	@State private(set) var user: [UserViewModel] = [UserViewModel]()
+	@State private var isLoading: Bool = false
+	@State private var isShowError: Bool = false
+	@State private var error: HomeErrorView?
+	@State private var isInCall = false
+	
+	let inspection = ViewInspector<Self>()
+
+	// MARK: - Body
 	var body: some View {
-		GeometryReader { _ in
-			NavigationView {
-				self.content
-					.navigationBarTitle("Home")
+		NavigationView {
+			GeometryReader { geometry in
+				ZStack {
+					HStack {
+						ListServerView(servers: $servers, isAddNewServer: $isAddNewServer, action: getServerInfo)
+						VStack(spacing: Constants.spacing) {
+							HStack {
+								Text(serverName)
+									.font(AppTheme.shared.fontSet.font(style: .display3))
+									.foregroundColor(titleColor)
+								Spacer()
+								ImageButton(AppTheme.shared.imageSet.menuIcon) {
+									withAnimation {
+										isShowMenu.toggle()
+									}
+								}
+								.foregroundColor(titleColor)
+							}
+							content
+								.padding(.top, Constants.padding)
+						}
+						.padding(Constants.padding)
+					}
+					.hiddenNavigationBarStyle()
+					.padding(.top, Constants.paddingTop)
+					.hideKeyboardOnTapped()
+					NavigationLink(destination: ChatView(inputStyle: .default, groupId: selectedGroupId, avatarLink: ""), isActive: $navigateToChat) {
+						EmptyView()
+					}
+					NavigationLink(
+						destination: LoginView(customServer: customServer, navigateToHome: navigateToLogin, rootIsActive: $navigateToLogin),
+						isActive: $navigateToLogin,
+						label: {
+						})
+				}
+				.hiddenNavigationBarStyle()
+				
+				if isShowMenu {
+					LinearGradient(gradient: Gradient(colors: colorScheme == .light ? AppTheme.shared.colorSet.gradientPrimary.compactMap({ $0.opacity(Constants.opacity) }) : AppTheme.shared.colorSet.gradientBlack), startPoint: .leading, endPoint: .trailing)
+						.blur(radius: Constants.blur)
+						.edgesIgnoringSafeArea(.vertical)
+					MenuView(isShowMenu: $isShowMenu, user: $user, chageStatus: { status in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/fe9f474e9dacdb67166a3b18715e917932515a0f/ClearKeep/Scenes/Home/HomeView.swift#L176"></span>Arguments should be either on the same line, or one per line.<br />
<code>multiline_arguments</code> <code>HomeView.swift:176</code></td>
    </tr>
  </tbody>
</table>